File: apzkr-pzpi-21-5-kravchenko-yevhenii/Task1-Server/BLL/Infrastructure/Models/User/UserProfileModel.cs
using BLL.Infrastructure.Models.UserMembership;
using Infrastructure.Enums;

namespace BLL.Infrastructure.Models;
public class UserProfileModel
{
    public int Id { get; set; }

    public string FirstName { get; set; } = null!;

    public string LastName { get; set; } = null!;

    public byte[]? ProfilePicture { get; set; }

    public string Address { get; set; } = null!;

    public string PhoneNumber { get; set; } = null!;

    public DateTime BirthDate { get; set; }

    public string BirthDateString { get; set; } = null!;

    public string Email { get; set; } = null!;

    public string Login { get; set; } = null!;

    public DateTime RegistrationDate { get; set; }

    public string RegistrationDateStr { get; set; } = null!;

    public Role Role { get; set; }

    public UserMembershipModel? UserMembershipModel { get; set; }
}

File: apzkr-pzpi-21-5-kravchenko-yevhenii/Task3-WebClient/src/app/core/models/user/user-profile-model.ts
import { Role } from "../../enums/role";
import { UserMembershipModel } from "../user-membership/user-membership-model";

export interface UserProfileModel {
    id: number,
    firstName: string,
    lastName: string,
    profilePicture: string,
    address: string,
    phoneNumber: string,
    birthDate: Date | null,
    birthDateString: string,
    email: string,
    login: string,
    registrationDate: Date | null,
    registrationDateStr: string,
    role: Role,
    userMembershipModel: UserMembershipModel | null
}
